File: frontend/src/http/continents.ts
export const fetchContinentRawData = (continentId: string): Promise<Response> => {
    return fetch('/api/continents/'+continentId+'/raw');
};

export const fetchContinentPredict = (continentId: string, year: number): Promise<Response> => {
    return fetch('/api/continents/'+continentId+'/predict/'+year);
};

File: continents.py
from flask_restful import Resource, Api
from helpers import continents_raw_data
import pickle
import numpy as np
import sklearn
from flask import request

class ContinentsRaw(Resource):
    def get(self, continentId):
        return continents_raw_data(continentId)

class ContinentsPredict(Resource):
    def get(self, continentId, year):
        loaded_model = pickle.load(open('models/continents/continents_model_region_'+continentId+'.sv', 'rb'))
        result = loaded_model.predict(np.array([year]).reshape(-1,1))
        return float(result[0])